pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -16,33 +29,31 @@ export class PyTerminalPlugin extends Plugin {
         this.app = app;
     }
 
-    configure(config: AppConfig & { terminal?: boolean | 'auto' }) {
+    configure(config: AppConfigStyle) {
         // validate the terminal config and handle default values
-        const t = config.terminal;
-        if (t !== undefined && t !== true && t !== false && t !== 'auto') {
-            const got = JSON.stringify(t);
-            throw new UserError(
-                ErrorCode.BAD_CONFIG,
-                'Invalid value for config.terminal: the only accepted' +
-                    `values are true, false and "auto", got "${got}".`,
-            );
-        }
+        const { terminal: t, docked: d } = config;
+        validate('terminal', 'auto', t);
+        validate('docked', 'docked', d);
         if (t === undefined) {
             config.terminal = 'auto'; // default value
         }
+        if (d === undefined) {
+            config.docked = 'docked'; // default value
+        }
     }
 
-    beforeLaunch(config: AppConfig & { terminal?: boolean | 'auto' }) {
+    beforeLaunch(config: AppConfigStyle) {
         // if config.terminal is "yes" or "auto", let's add a <py-terminal> to
         // the document, unless it's already present.
-        const t = config.terminal;
-        if (t === true || t === 'auto') {
-            if (document.querySelector('py-terminal') === null) {
-                logger.info('No <py-terminal> found, adding one');
-                const termElem = document.createElement('py-terminal');
-                if (t === 'auto') termElem.setAttribute('auto', '');
-                document.body.appendChild(termElem);
-            }
+        const { terminal: t, docked: d } = config;
+        const auto = t === true || t === 'auto';
+        const docked = d === true || d === 'docked';
```

Comment:
question:
what happens if I write `<py-terminal docked>` (without specifying a value for the attribute)?
What is the value of `d` in this case?

Reply:
I'have copied the dance form `auto` logic and, unless I have missed anything, when defined as config it's either `true` or `"docked"`, that's what the validation is used for in the `constructor`, no other possible values, while if present on the layout and not passed as config the `isDocked()` method will return true.

Is there any particular reason you are sure about how `auto` behave but not `docked` ? have I missed some bootstrap logic for elements already present on the page? 🤔

Reply:
no, I just forgot how it works 😅